snapshot/snapshot.go

```diff
@@ -10,96 +10,79 @@ import (
 	"github.com/lyft/goruntime/snapshot/entry"
 )
 
-func min(lhs uint64, rhs uint64) uint64 {
-	if lhs < rhs {
-		return lhs
-	} else {
-		return rhs
-	}
-}
-
-// Random number generator. Implementations should be thread safe.
-type RandomGenerator interface {
-	// @return uint64 a new random number.
-	Random() uint64
-}
-
-// Implementation of RandomGenerator that uses a time seeded random generator.
-type randomGeneratorImpl struct {
-	sync.Mutex
-	random *rand.Rand
-}
-
-func (r *randomGeneratorImpl) Random() uint64 {
-	r.Lock()
-	v := uint64(r.random.Int63())
-	r.Unlock()
-	return v
+type random struct {
+	mu sync.Mutex
+	rr *rand.Rand
 }
 
-var defaultRandomGenerator RandomGenerator = &randomGeneratorImpl{
-	random: rand.New(rand.NewSource(time.Now().UnixNano())),
+func (r *random) Uint64() uint64 {
+	r.mu.Lock()
+	x := r.rr.Int63()
+	r.mu.Unlock()
+	return uint64(x)
 }
 
 // Implementation of Snapshot for the filesystem loader.
 type Snapshot struct {
 	entries map[string]*entry.Entry
+	rand    random
 }
 
 func New() (s *Snapshot) {
-	s = &Snapshot{
+	rr := random{
+		rr: rand.New(rand.NewSource(time.Now().UnixNano())),
+	}
+	return &Snapshot{
 		entries: make(map[string]*entry.Entry),
+		rand:    rr,
 	}
+}
 
-	return
+func min(lhs, rhs uint64) uint64 {
+	if lhs <= rhs {
+		return lhs
+	}
+	return rhs
 }
 
 func (s *Snapshot) FeatureEnabled(key string, defaultValue uint64) bool {
-	return defaultRandomGenerator.Random()%100 < min(s.GetInteger(key, defaultValue), 100)
+	return s.rand.Uint64()%100 < min(s.GetInteger(key, defaultValue), 100)
```

Comment:
Should we use `Intn(100)` for readability? Is that the same thing?

Reply:
Yes.